### DISCUSSION TITLE
Empty list of Service Instances

### DISCUSSION THREAD

**Discussion by jarofi** [apache/skywalking#11250] [Q&A] [answered]
We are using Native UI V9.5.0 and we encounter a problem with displaying Service Instances on the standard General-Service dashboard. If a short history interval is selected (from the default 30 min to a few hours/days), the list of instances is empty, even though there is data available in the selected interval (I can access it by extending the interval). How can I ensure that the instances are always displayed even if the default interval is set to 30 min? Thank you in advance for your help.

**Comment by wu-sheng** [comment]
This is not a hard thing to debug. There is no quick answer if you don't ever debug and provide more context.

**Comment by weixiang1862** [comment]
What storage backend are you using? We fix a jdbc instance query problem recently #11024. If you are using mysql or others, you can validate it by `ghcr.io/apache/skywalking/oap:f1db25560521534a1599350e9101c4fd3ab52c7e`(It is not release version, DO NOT USE FOR PRODUCT), or you should try debug yourself and provide more details.

**Comment by jarofi** [comment]
@wu-sheng @weixiang1862 Thank you for your response. I'm sorry, but I'm just getting started with SkyWalking. I tried debugging with swctl tool:
**command**
`swctl --debug instance list --start="2023-08-21" --end="2023-08-23" --service-name "DEMO|Tomcat10|DEVEL"`

**result**
```
DEBUG Debug mode is enabled
DEBUG Using configurations:
<...omitted...>
DEBUG Start time: 2023-08-21 end time: 2023-08-23 timezone:
DEBUG >> variables: map[]
DEBUG >> query:
query {
    result: version
}
DEBUG >> headers: map[Accept:[application/json; charset=utf-8] Content-Type:[application/json; charset=utf-8]]
DEBUG << {"data":{"result":"9.5.0-d47165d"}}
DEBUG >> variables: map[serviceName:DEMO|Tomcat10|DEVEL]
DEBUG >> query:
query findService($serviceName: String!) {
    result: findService(serviceName: $serviceName) {
        id name group shortName layers normal
    }
}
DEBUG >> headers: map[Accept:[application/json; charset=utf-8] Content-Type:[application/json; charset=utf-8]]
DEBUG << {"data":{"result":{"id":"REVNT3xUb21jYXQxMHxERVZFTA==.1","name":"DEMO|Tomcat10|DEVEL","group":"","shortName":"DEMO|Tomcat10|DEVEL","layers":["GENERAL"],"normal":true}}}
DEBUG service-id=REVNT3xUb21jYXQxMHxERVZFTA==.1, service-name=DEMO|Tomcat10|DEVEL
DEBUG >> variables: map[]
DEBUG >> query:
query {
    result: version
}
DEBUG >> headers: map[Accept:[application/json; charset=utf-8] Content-Type:[application/json; charset=utf-8]]
DEBUG << {"data":{"result":"9.5.0-d47165d"}}
DEBUG >> variables: map[duration:{2023-08-21 2023-08-23 DAY} serviceId:REVNT3xUb21jYXQxMHxERVZFTA==.1]
DEBUG >> query:

query ($serviceId: ID!, $duration: Duration!) {
    result: listInstances(duration: $duration, serviceId: $serviceId) {
        id
        name
        language
        instanceUUID
        attributes {
            name
            value
        }
    }
}
DEBUG >> headers: map[Accept:[application/json; charset=utf-8] Content-Type:[application/json; charset=utf-8]]
DEBUG << {"data":{"result":[]}}
[]
```
@weixiang1862 We use PostreSQL as storage solution. So it's quite likely that the bug #11024 affects us. But we don't use Docker, so I would like to ask if I can verify this in another way?

**Reply by wu-sheng** [reply]
You could build the binary from the source codes. Ref https://skywalking.apache.org/docs/main/next/en/guides/how-to-build/

**Comment by jarofi** [comment, marked as answer]
@wu-sheng @weixiang1862 Thank you very much for your help.
